silkworm/node/db/access_layer.cpp

```diff
@@ -1308,6 +1308,29 @@ bool DataModel::read_rlp_transactions(BlockNum height, const evmc::bytes32& hash
     return read_rlp_transactions_from_snapshot(height, transactions);
 }
 
+std::optional<BlockNum> DataModel::read_tx_lookup(const evmc::bytes32& tx_hash) const {
+    auto block_num = read_tx_lookup_from_db(tx_hash);
+    if (block_num) {
+        return block_num;
+    }
+
+    return read_tx_lookup_from_snapshot(tx_hash);
+}
+
+std::optional<BlockNum> DataModel::read_tx_lookup_from_db(const evmc::bytes32& tx_hash) const {
+    auto cursor = txn_.ro_cursor(table::kTxLookup);
+    auto data{cursor->find(to_slice(tx_hash.bytes), /*throw_notfound = */ false)};
+    if (!data) {
+        return std::nullopt;
+    }
+    auto block_num = endian::load_big_u64(static_cast<const unsigned char*>(data.value.data()));
```

Comment:
Here you can use our `from_slice` utility function to avoid spreading the cast all over the places like this:
```
return endian::load_big_u64(from_slice(data.value).data());
```

Reply:
done